### PR TITLE
✅ Aggregate coverage during testing and report after

### PR DESCRIPTION
**Pull request by rcebulko**
Hooks into the `istanbul-middleware/lib/core` module to directly merge coverage objects extracted from the page after each test, and reports them all at once at the end of the e2e test suite.

### CODE REVIEW

File: build-system/tasks/e2e/describes-e2e.js [hunk 1]
```diff
@@ -335,38 +339,43 @@ class ItConfig {
...
-        (res) => {
-          res.on('data', () => {});
-          res.on('end', resolve);
-          res.on('error', reject);
-        }
-      );
-      req.write(JSON.stringify(cov));
-      req.end();
-    });
+async function collectCoverage(env) {
```
Comment (rsimha) [inline build-system/tasks/e2e/describes-e2e.js:346]: Naming nit: Would it make sense to call this function something like `updateCoverage` or `appendCoverage` to clearly indicate that it's going to incrementally add to existing data and not re-collect everything?

**Review by rsimha — COMMENTED** [review]
LGTM overall. Will let @estherkim approve since she knows this code better than I do.

File: build-system/tasks/e2e/describes-e2e.js [hunk 2]
```diff
@@ -335,38 +339,43 @@ class ItConfig {
...
   }
 }
 
+/**
+ * Reports code coverage data to an aggregating endpoint.
+ * @return {Promise<void>}
+ */
+async function reportCoverage() {
+  const coverage = getCoverageObject();
+  await new Promise((resolve, reject) => {
```
Comment (estherkim) [inline build-system/tasks/e2e/describes-e2e.js:359]: Can this be an async function instead of constructing a promise?

**Review by estherkim — APPROVED** [review]
Nice improvement 👍 

p.s @rsimha thanks for pulling me in but it's been a while since e2e tests were introduced, I should think that we're all familiar with it now :)

File: build-system/tasks/e2e/describes-e2e.js (hunk 2 above)
Comment (rcebulko) [reply build-system/tasks/e2e/describes-e2e.js]: Updated to use `fetch`

File: build-system/tasks/e2e/describes-e2e.js (hunk 1 above)
Comment (rcebulko) [reply build-system/tasks/e2e/describes-e2e.js]: Done

**Pull request merged**